### PR TITLE
PHP 8.3 build

### PR DESCRIPTION
**Pull request by michael-rubel**
## About

Updates constraints for PHP 8.3.

### CODE REVIEW

**Comment by michael-rubel** [comment]
Ah, there are more dependencies to update. Probably worth waiting till a stable release. 🙂

**Comment by michael-rubel** [comment]
Hi @Ocramius 

Left the PR in [better-reflection](https://github.com/Roave/BetterReflection/pull/1381). Could you take a look?

**Comment by Ocramius** [comment]
@michael-rubel upstream is released BTW  - https://github.com/Roave/BetterReflection/releases/tag/6.18.0

**Comment by michael-rubel** [comment]
@Ocramius the next one. 😁

https://github.com/Roave/Signature/pull/64

**Comment by Ocramius** [comment]
@michael-rubel also released 👍

**Comment by Ocramius** [comment]
Seems like the `roave/better-reflection` upstream dependency bump introduced some type alignment issues:

```
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:14:63: MissingImmutableAnnotation: Roave\BetterReflection\SourceLocator\Located\LocatedSource is marked @psalm-immutable, but Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory is not marked @psalm-immutable (see https://psalm.dev/213)
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:22:5: MissingImmutableAnnotation: Roave\BetterReflection\SourceLocator\Located\LocatedSource::getSource is marked @psalm-immutable, but Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory::getSource is not marked @psalm-immutable (see https://psalm.dev/213)
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:27:5: MissingImmutableAnnotation: Roave\BetterReflection\SourceLocator\Located\LocatedSource::getName is marked @psalm-immutable, but Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory::getName is not marked @psalm-immutable (see https://psalm.dev/213)
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:32:5: MissingImmutableAnnotation: Roave\BetterReflection\SourceLocator\Located\LocatedSource::getFileName is marked @psalm-immutable, but Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory::getFileName is not marked @psalm-immutable (see https://psalm.dev/213)
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:32:36: MoreSpecificReturnType: The declared return type 'non-empty-string|null' for Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory::getFileName is more specific than the inferred return type 'null|string' (see https://psalm.dev/070)
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:40:16: LessSpecificReturnStatement: The type 'string' is more general than the declared return type 'non-empty-string|null' for Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory::getFileName (see https://psalm.dev/129)
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:43:5: MissingImmutableAnnotation: Roave\BetterReflection\SourceLocator\Located\LocatedSource::isInternal is marked @psalm-immutable, but Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory::isInternal is not marked @psalm-immutable (see https://psalm.dev/213)
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:48:5: MissingImmutableAnnotation: Roave\BetterReflection\SourceLocator\Located\LocatedSource::getExtensionName is marked @psalm-immutable, but Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory::getExtensionName is not marked @psalm-immutable (see https://psalm.dev/213)
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:53:5: MissingImmutableAnnotation: Roave\BetterReflection\SourceLocator\Located\LocatedSource::isEvaled is marked @psalm-immutable, but Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory::isEvaled is not marked @psalm-immutable (see https://psalm.dev/213)
Error: src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php:58:5: MissingImmutableAnnotation: Roave\BetterReflection\SourceLocator\Located\LocatedSource::getAliasName is marked @psalm-immutable, but Roave\BackwardCompatibility\SourceLocator\LocatedSourceWithStrippedSourcesDirectory::getAliasName is not marked @psalm-immutable (see https://psalm.dev/213)
```

I think we should revert the `composer.lock` changes, and only update the `composer.lock` hash (via `composer update nothing`).

Any issues arising from dependency upgrades would then be fixed later on, IMO, in a separate patch, unless it's work you're willing to do here.

**Comment by Ocramius** [comment]
```
Verifying lock file contents can be installed on current platform.
- Required package "roave/better-reflection" is in the lock file as "6.5.0" but that does not satisfy your constraint "^6.18.0".
```

We need to revert also the dependency bumps in `composer.json`, for now :D

File: composer.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
         "nikic/php-parser": "^4.15.3",
         "nikolaposa/version": "^4.1.0",
         "ocramius/package-versions": "^2.7.0",
-        "roave/better-reflection": "^6.18.0",
+        "roave/better-reflection": "^6.5.0",
```
Comment (Ocramius) [inline composer.json]: With this commit, the hash should be changed again :D

**Comment by michael-rubel** [comment]
@Ocramius Yeah, seems safe now. 😁

**Review by Ocramius — APPROVED** [review]
Thanks @michael-rubel!

**Comment by Ocramius** [comment]
And released :-)

**Pull request merged**